dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,255 @@
+
+            # Compute the delta speed components along the (x, y) axis
+            speed_x += delta_x
+            speed_y += delta_y
+
+            # Take a step along velocity.
+            x += speed_x
+            y += speed_y
+            # Bounce off edges.
+            for j in range(self.num_digits):
+                if x[j] <= 0:
+                    x[j] = 0
+                    speed_x[j] = -speed_x[j]
+                elif x[j] >= 1.0:
+                    x[j] = 1.0
+                    speed_x[j] = -speed_x[j]
+                if y[j] <= 0:
+                    y[j] = 0
+                    speed_y[j] = -speed_y[j]
+                elif y[j] >= 1.0:
+                    y[j] = 1.0
+                    speed_y[j] = -speed_y[j]
+            trajectory_x[frame_id] = x
+            trajectory_y[frame_id] = y
+
+        # Scale to the size of the canvas.
+        trajectory_x = (canvas_size[1] * trajectory_x).astype(np.int32)
+        trajectory_y = (canvas_size[0] * trajectory_y).astype(np.int32)
+        return trajectory_x, trajectory_y
+
+    def _get_sequence(self, verbose=False):
+        trajectory_x, trajectory_y = self._get_random_trajectory()
+
+        # minibatch data
+        if self.random_background:
+            out_sequence = self._rng.rand(self.seq_length+1,
+                                          self.frame_size[0],
+                                          self.frame_size[1], 1)
+        else:
+            out_sequence = np.zeros((self.seq_length+1,
+                                     self.frame_size[0],
+                                     self.frame_size[1], 1),
+                                    dtype=np.float32)
+
+        for digit_id in range(self.num_digits):
+
+            # get random digit from dataset
+            curr_data_idx = self._rng.randint(
+                0, self._MNIST_data.shape[0]-1)
+            digit_image = self._MNIST_data[curr_data_idx]
+
+            zoom_factor = 1
+            if self.digits_sizes != 28:
+                zoom_factor = int(self.digits_sizes[digit_id]/28)
+                digit_image = zoom(digit_image, zoom_factor)
+            digit_size = digit_image.shape[0]
+
+            # generate video
+            digit_image = np.expand_dims(digit_image, -1)
+            import ipdb; ipdb.set_trace()
+            for i in range(self.seq_length+1):
+                top = trajectory_y[i, digit_id]
+                left = trajectory_x[i, digit_id]
+                bottom = top + digit_size
+                right = left + digit_size
+                out_sequence[i, top:bottom, left:right, :] = np.maximum(
+                    out_sequence[i, top:bottom, left:right, :], digit_image)
+        return out_sequence
+
+    def _reset(self, *args, **kwargs):
+        """ This function is used to reset the rng and data index from
+        where the digits are sampled to create the video sequence
+        """
+        if self.which_set != 'train':
```

Comment:
If the rng is not reset we have potentially an "infinite" set for the training. Right?